### PR TITLE
Fixed data job status in case of OOM

### PR DESCRIPTION
**Pull request by mivanov1988**
# Why
Currently, in the case of Data Job failure caused by OOM, the Control Services marks the data job with User Error but after some time it overrides the status to Platform Error. This is caused by K8S Job events that come after the data job completion. They don't contain enough data and the Control Service treats them as Platform Errors.

Log of the data job execution:

```
Jan 27 14:38:38 tpcs-dep-68f96bc974-qhqf4 [thread-5] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=BackoffLimitExceeded, succeeded=false, opId=01674829787797, startTime=2023-01-27T14:29:48Z, endTime=2023-01-27T14:32:28Z, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=null)
source: 10.216.162.76 event_type: v4_254d2c9c facility: user priority: debug hostname: tpcs-dep-68f96bc974-qhqf4

Jan 27 14:33:24 tpcs-dep-68f96bc974-8jz5h [thread-2] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=BackoffLimitExceeded, succeeded=false, opId=01674829787797, startTime=2023-01-27T14:29:48Z, endTime=2023-01-27T14:32:28Z, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=null)
source: 10.216.162.116 event_type: v4_254d2c9c facility: user priority: debug hostname: tpcs-dep-68f96bc974-8jz5h

Jan 27 14:32:29 tpcs-dep-68f96bc974-qhqf4 [thread-1] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=BackoffLimitExceeded, succeeded=false, opId=01674829787797, startTime=2023-01-27T14:29:48Z, endTime=2023-01-27T14:32:28Z, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=OOMKilled)
source: 10.216.162.76 event_type: v4_254d2c9c facility: user priority: debug hostname: tpcs-dep-68f96bc974-qhqf4

Jan 27 14:32:28 tpcs-dep-68f96bc974-qhqf4 [thread-1] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=BackoffLimitExceeded, succeeded=false, opId=01674829787797, startTime=2023-01-27T14:29:48Z, endTime=2023-01-27T14:32:28Z, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=OOMKilled)
source: 10.216.162.76 event_type: v4_254d2c9c facility: user priority: debug hostname: tpcs-dep-68f96bc974-qhqf4

Jan 27 14:29:48 tpcs-dep-68f96bc974-qhqf4 [thread-1] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=null, succeeded=null, opId=01674829787797, startTime=2023-01-27T14:29:48Z, endTime=null, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=null)
source: 10.216.162.76 event_type: v4_254d2c9c facility: user priority: debug hostname: tpcs-dep-68f96bc974-qhqf4

Jan 27 14:29:48 tpcs-dep-68f96bc974-qhqf4 [thread-1] com.vmware.taurus.service.monitoring.DataJobMonitor Storing Data Job execution status: KubernetesService.JobExecution(executionId=miroslavi-test44-1674829787, executionType=manual, jobName=miroslavi-test44, podTerminationMessage=null, jobTerminationReason=null, succeeded=null, opId=01674829787797, startTime=null, endTime=null, jobVersion=355c5077da74ecf3623e03b7cd17996faf9702b5, jobSchedule=11 23 5 8 1, resourcesCpuRequest=0.5, resourcesCpuLimit=1.0, resourcesMemoryRequest=1000, resourcesMemoryLimit=4000, deployedDate=2023-01-27T14:28:37.088538Z, deployedBy=miroslavi, containerTerminationReason=null)
```
# What
Implemented a logic wich omits events that come after the Data Job completion.

# Testing done
Introduced integration test.

Signed-off-by: Miroslav Ivanov [miroslavi@vmware.com](mailto:miroslavi@vmware.com)

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/DataJobDeploymentExtension.java [hunk 1]
```diff
@@ -124,7 +126,8 @@ public void beforeEach(ExtensionContext context) throws Exception {
 
     if (!initialized) {
       byte[] jobZipBinary =
-          IOUtils.toByteArray(getClass().getClassLoader().getResourceAsStream(jobSource));
+          IOUtils.toByteArray(
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/DataJobDeploymentExtension.java:129]: Those zip files are a bit hard to review. I've been thinking it might be good idea to keep them as a directory and zip them here.

This is out of scope for this change and it would be better done in separate PR anyhow, just wondering what do you think.

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobOOMIT.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = ControlplaneApplication.class)
+public class DataJobOOMIT extends BaseIT {
+
+  @RegisterExtension
+  static DataJobDeploymentExtension dataJobDeploymentExtension =
+      new DataJobDeploymentExtension("oom_job.zip");
+
+  @Test
+  public void testDataJob_causesOOM_shouldCompleteWithUserError(
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobOOMIT.java:31]: Did you verify the test fails without the fix (the change in KuberntesService). 

I am wondering - 
If the error happens after the job is finished. Shouldn't we wait for some condition for the status to change ?

Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobOOMIT.java:31]: Yes, I did with another JobExecutionUtil.checkDataJobExecutionStatus(
                executionId,
                DataJobExecution.StatusEnum.PLATFORM_ERROR,
                opId,
                jobName,
                teamName,
                username,
                mockMvc);

However, it takes more than 5 minutes and it's not consistent.

**Review by dakodakov — APPROVED** [review]

**Review by tozka — APPROVED** [review]
